lib/Controller/Schedule.php

```diff
@@ -773,6 +775,15 @@ public function addForm(Request $request, Response $response, ?string $from, ?in
         $defaultLat = (float)$this->getConfig()->getSetting('DEFAULT_LAT');
         $defaultLong = (float)$this->getConfig()->getSetting('DEFAULT_LONG');
 
+        // Dispatch an event to retrieve criteria for scheduling from the OpenWeatherMap connector.
+        $event = new ScheduleCriteriaRequestEvent();
+        $this->getDispatcher()->dispatch($event, ScheduleCriteriaRequestEvent::$NAME);
+
+        // Retrieve the data from the event
+        $criteria = $event->getCriteria();
+
+        // maybe add a scheduleCriteria and add that in the $addFormData instead of adding the type, metric and value individually
```

Comment:
Maybe? I think you have done this?

Reply:
Sorry about this one. This was actually a note for myself while I was still working on the logic. I was supposed to remove it, but it seems I forgot to do so.